app/controllers/goals_controller.rb

```diff
@@ -28,6 +28,11 @@ def update
     end
   end
 
+  def destroy
+    @goal.destroy
+    redirect_to user_goals_path(@user)
```

Comment:
what happens if destroy operation fails? i think we should handle that case as well

Reply:
also we can add authorization to check whether the current user is allowed to destroy the goal or not

Reply:
add proper tests after addressing the above mentioned comments

Reply:
thank you i added the edge case and the spec regarding it